File: acme-java-springboot/src/main/java/acme/entity/Transaction.java
package acme.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Version;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;
import java.util.UUID;

@Table("transaction")
public record Transaction(
    @Id UUID transactionId,
    UUID fromAccountId,
    UUID toAccountId,
    Long amount,
    LocalDateTime timestamp,
    String status,
    @Version Long version
) {
    public Transaction(UUID transactionId, UUID fromAccountId, UUID toAccountId,
                       Long amount, LocalDateTime timestamp, String status) {
        this(transactionId, fromAccountId, toAccountId, amount, timestamp, status, 0L);
    }
}

File: acme-springboot-nonblocking/src/main/kotlin/acme/entity/Transaction.kt
package acme.entity

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Version
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime
import java.util.*

@Table("transaction")
data class Transaction(
    @Id
    val transactionId: UUID?,
    val fromAccountId: UUID,
    val toAccountId: UUID,
    val amount: Long,
    val timestamp: LocalDateTime,
    val status: String,
    @Version
    val version: Long = 0
)
